Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/ThresholdedRandomCutForest.java

```diff
@@ -170,14 +171,31 @@ public ThresholdedRandomCutForest(ForestMode forestMode, TransformMethod transfo
         this.lastAnomalyDescriptor = descriptor;
     }
 
-    // for conversion from other thresholding models
-    public ThresholdedRandomCutForest(RandomCutForest forest, double futureAnomalyRate, List<Double> values) {
+    // this constructor produces an internally shingled ThresholdedRCF model from an
+    // externally shingled ThresholdedRCF model (which may or may not be in use in
+    // older
+    // versions of OpenSearch) absent any transformations and augmentations.
+    // A benefit of this conversion would be that imputations would be accessible
+    // to ThresholdedRCF -- that is, even if not every value of the input tuple is
+    // known
+    // the function process() would be able to provide an anomaly score (which is
+    // likely near
+    // minimum, since RCF is used to fill in the missing values). As a result, high
+    // values of the
+    // anomaly score will continue to be likely anomalies.
+    // Note that the basic RandomCutForest cannot be changed easily
+    // but the process() function would only require a fraction of the input
+    // see ThresholdedRandomCutForestMapperTest
+    public ThresholdedRandomCutForest(RandomCutForest forest, double futureAnomalyRate, List<Double> values,
+            double[] lastShingledInput) {
         this.forest = forest;
         int dimensions = forest.getDimensions();
-        int inputLength = (forest.isInternalShinglingEnabled()) ? dimensions / forest.getShingleSize()
-                : forest.getDimensions();
+
+        int inputLength = dimensions / forest.getShingleSize();
```

Comment:
We are currently operating under the assumption that the input to the RandomCutForest is internally shingled. However, this assumption could disrupt Anomaly Detection (AD) functionalities, as we utilize this constructor to maintain backward compatibility with the single-stream AD that relies on external shingling. To address this issue in AD, we would need to undertake a comprehensive overhaul, including redoing the cold start process and modifying all areas related to external shingling (such as imputation). However, this is not a straightforward change and would require a significant amount of work.

Reply:
No no .. we do not change the random cut forest -- because any change could affect the cuts and corrupt the trees. To change the RCF we are better of extracting all points and then rebuilding them. But for RCFs external and internal shingling executes the same code and would perform opportunistic compression.  The forest is NOT changed -- 

this.forest = forest;

If this discussion is holding the PR up, then let's proceed. Note that a similar function existed earlier. This is an improvement and we should move on. What this does is *externally shingled TRCF* is converted to *internally shingled TRCF* and RCF is managed on its own -- the user should stop worrying about that. But if that is still a concern then this function should not be used, period. This is why I said this *may/may not* be helpful based on internals of what exists in prod currently.

Reply:
Ok - I think I see the disconnect. Changing the comment.